rusk/src/lib/http/rusk.rs

```diff
@@ -134,6 +140,33 @@ impl Rusk {
         let crs = rusk_profile::get_common_reference_string()?;
         Ok(ResponseData::new(crs).with_header("crs-hash", CRS_17_HASH))
     }
+
+    fn simulate_tx(&self, tx: &str) -> anyhow::Result<ResponseData> {
+        let tx = hex::decode(tx).map_err(|e| {
+            anyhow::anyhow!("Failed to decode transaction: {e}")
+        })?;
+        let tx = Transaction::from_slice(&tx)
+            .map_err(|e| anyhow::anyhow!("Invalid Data: {e:?}"))?;
+        if tx.gas_limit() > self.get_block_gas_limit() {
+            return Err(anyhow::anyhow!("Gas limit is too high."));
+        }
+        let mut session = self.query_session(None).map_err(|e| {
+            anyhow::anyhow!("Failed to initialize a session: {e:?}")
+        })?;
+        let config = ExecutionConfig {
+            gas_per_deploy_byte: self.gas_per_deploy_byte(),
+            min_deploy_gas_price: self.min_deployment_gas_price(),
+            min_deploy_points: self.min_deploy_points(),
+            with_public_sender: false,
+        };
```

Comment:
Replace this with `self.vm_config.to_execution_config(block_height)` otherwise the `with_public_sender` is wrong